.github/workflows/rust-prs.yml

```diff
@@ -0,0 +1,459 @@
+    steps:
+      - name: Get sources
+        uses: actions/checkout@v4
+
+      - name: Build assets
+        uses: ./.github/actions/cli_assets
+
+      - name: Copy ~/.grafbase for upload
+        shell: bash
+        run: |
+          mkdir home-dot-grafbase
+          cp -r ~/.grafbase/* home-dot-grafbase/
+
+      - uses: actions/upload-artifact@v4
+        with:
+          name: home-dot-grafbase
+          path: home-dot-grafbase
+          retention-days: 5
+
+      - uses: actions/upload-artifact@v4
+        with:
+          name: cli-app-dist
+          path: packages/cli-app/dist
+          retention-days: 5
+
+      - uses: actions/upload-artifact@v4
+        with:
+          name: wrappers-dist.js
+          path: cli/wrappers/dist.js
+          retention-days: 5
+
+  builds:
+    needs: [what-changed, example-component, cli-assets]
+    if: |
+      (!cancelled())
+      && fromJson(needs.what-changed.outputs.rust).changed-packages
+      && (needs.example-component.result == 'skipped' || needs.example-component.result == 'success')
+      && (needs.cli-assets.result == 'skipped' || needs.cli-assets.result == 'success')
+    strategy:
+      fail-fast: false
+      matrix:
+        platform:
+          [
+            {
+              'target': 'x86_64-unknown-linux-musl',
+              'runner': 'ubicloud-standard-8'
+            },
+            {
+              'target': 'aarch64-unknown-linux-musl',
+              'runner': 'ubicloud-standard-8-arm'
+            },
+            {
+              'target': 'aarch64-apple-darwin',
+              'runner': 'macos-latest-xlarge'
+            },
+            {
+              'target': 'x86_64-apple-darwin',
+              'runner': 'macos-latest-xlarge'
+            },
+            {
+              'target': 'x86_64-pc-windows-msvc',
+              'runner': 'windows-latest-8-cores'
+            }
+          ]
+    runs-on: ${{ matrix.platform.runner }}
+    steps:
+      - name: Get sources
+        uses: actions/checkout@v4
+
+      - name: Dump inputs for debugging
+        shell: bash
+        run: |
+          echo ${{ needs.what-changed.outputs.rust }}
+
+      - name: Extract the Rust version to use from the `rust-toolchain.toml` file
+        shell: bash
+        run: |
+          rust_version=$(grep "channel" rust-toolchain.toml | cut -d "=" -f 2 | cut -d " " -f 2)
+          echo "RUST_VERSION=$rust_version" >> $GITHUB_ENV
+
+      - name: Install Rust
+        uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: ${{ env.RUST_VERSION }}
+          components: clippy, rustfmt
+
+      - name: Install musl-tools
+        if: ${{ contains(matrix.platform.target, 'linux') }}
+        shell: bash
+        run: |
+          sudo apt-get install musl musl-tools
+          # This seems like a horrible hack that might come back to bite, but lets see!
+          sudo ln -s /bin/g++ /bin/musl-g++
+          sudo ln -s /bin/g++ /bin/aarch64-linux-musl-g++
```

Comment:
😱 

We should consider just getting rid of musl eventually... Maybe consider having a certain glibc version minimum we require, document it and avoid having other dynamic dependencies, such as openssl et.al.

Reply:
Yeah, would definitely be nice.